### PR TITLE
Don't add cluster router routees on MemberPreparingForShutdown events

### PR DESCRIPTION
**Pull request by johanandren**
References #30329

### CODE REVIEW

File: akka-cluster/src/main/scala/akka/cluster/routing/ClusterRouterConfig.scala [hunk 1]
```diff
@@ -524,6 +524,9 @@ private[akka] trait ClusterRouterActor { this: RouterActor =>
     satisfiesRoles(m.roles) &&
     (settings.allowLocalRoutees || m.address != cluster.selfAddress)
 
+  private def memberShuttingDown(event: MemberEvent): Boolean =
+    event.isInstanceOf[MemberPreparingForShutdown] || event.isInstanceOf[MemberReadyForShutdown]
```
Comment (patriknw) [inline akka-cluster/src/main/scala/akka/cluster/routing/ClusterRouterConfig.scala:528]: How does this work for the typed routers? That is via the Receptionist.

File: akka-cluster/src/main/scala/akka/cluster/routing/ClusterRouterConfig.scala [hunk 2]
```diff
@@ -524,6 +524,9 @@ private[akka] trait ClusterRouterActor { this: RouterActor =>
     satisfiesRoles(m.roles) &&
     (settings.allowLocalRoutees || m.address != cluster.selfAddress)
 
+  private def memberShuttingDown(event: MemberEvent): Boolean =
```
Comment (patriknw) [inline akka-cluster/src/main/scala/akka/cluster/routing/ClusterRouterConfig.scala:527]: Why isn't `isAvailable` enough? It checks status Up or WeaklyUp. I thought the status would be `PreparingForShutdown` or `ReadyForShutdown` for those new events?

Comment (johanandren) [reply akka-cluster/src/main/scala/akka/cluster/routing/ClusterRouterConfig.scala:527]: Indeed, I may have trusted the bug report a bit too blindly, it should work out of the box.

File: akka-cluster/src/main/scala/akka/cluster/routing/ClusterRouterConfig.scala (hunk 1 above)
Comment (johanandren) [reply akka-cluster/src/main/scala/akka/cluster/routing/ClusterRouterConfig.scala:528]: Yeah, the receptionist provides a separate lists for reachable and then a list with all routees, the group router uses the list of reachable unless that is empty, then falls back to the full list. I guess we could improve that with logic like what exists here.

**Comment by johanandren** [comment]
Closing this as it seems it was not actually the/a problem to start with

**Pull request closed without merging**